src/config/context/model.ts

```diff
@@ -0,0 +1,58 @@
+import { injectable, registry, } from 'tsyringe';
+import fs from 'fs';
+import { CONTEXTFILE_PATH } from '../../constants';
+
+export interface IContext {
+  readonly current?: string,
+  readonly store: {
+    [name: string]: string
+  }
+}
+
+export class Context implements IContext {
+  current?: string | undefined;
+  store: { [name: string]: string; };
+
+  constructor(ctx: IContext) {
+    this.current = ctx.current;
+    this.store = ctx.store;
+  }
+
+  getContext(contextName: string) {
+    return this.store[contextName as string];
+  }
+}
+
+export interface IContextAllocator {
+  contextFilePath?: string
+  load: () => Context | undefined
+  save: (context: Context) => boolean
+}
+
+@injectable()
+@registry([
+  {
+    token: 'IContextAllocator',
+    useToken: ContextAllocator
+  }
+])
+export class ContextAllocator implements IContextAllocator {
+  contextFilePath = CONTEXTFILE_PATH;
+  load(): Context | undefined {
+    try {
+      return new Context(JSON.parse(fs.readFileSync(this.contextFilePath, 'utf8')) as IContext);
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  save(context: Context) {
+    try {
+      fs.writeFileSync(this.contextFilePath, JSON.stringify(context), { encoding: 'utf8' });
+      return true;
+    } catch (error) {
+      return false;
```

Comment:
Why do you return `boolean` value, not undefined/throw Error?

Reply:
Would it be a good idea to return the updated context if successful and undefined if unsuccessful?

Reply:
In case when you load context, return context/undefined is good approach, but for saving I think that it will be better when you return, as you mentioned, also context/undefined, but please make `console.warn` in the catch case, because user should know that he/she cannot save context, e.g. due to fact that hasn't permission. Please also use the `console.warn` (with error information) in the catch of `load` function.

Reply:
The lint is shouting for using `console.warn` also I was thinking that currently, the starting point is in the `cli.ts` file `render(commandsRouter(cli));`  Could we wrap it in an async function and add a catch clause there hooking another ink component for all the Error messages there. So in the whole code we can just throw error messages handle all of them in one single place.

Reply:
Even `console.info`, or `console.error`? I have seen the latest changes - we can go with `Context | undefined`, but at least give some feedback to users.

> The lint is shouting for using console.warn 

We can also change linter rules if we need. For CLI app we can enable some `console.*` methods.

> So in the whole code we can just throw error messages handle all of them in one single place.

You could have several errors at once in one command and I would prefer to buffer these errors and shown at the end - but this should be implemented in the future.

Reply:
>You could have several errors at once in one command and I would prefer to buffer these errors and shown at the end - but this should be implemented in the future.

could work on this in the next PR.

Reply:
> could work on this in the next PR.

Make for it issue, really, because it can sound easy, but it probably won't 😅